extensions/podman/src/extension.ts

```diff
@@ -1054,24 +1060,39 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
           command: res.fixCommand,
         };
         warnings.push(warning);
+        if (!res.successful) {
+          telemetryRecords[res.name] = res.description ? res.description : res.name;
+        }
       }
 
       extensionApi.context.setValue('requirementsStatus', successful ? 'ok' : 'failed', 'onboarding');
       extensionApi.context.setValue('warningsMarkdown', warnings, 'onboarding');
+      telemetryLogger.logUsage('podman.onboarding.checkPodmanRequirements', telemetryRecords);
     },
   );
 
   const onboardingInstallPodmanCommand = extensionApi.commands.registerCommand(
     'podman.onboarding.installPodman',
     async () => {
+      let installation: InstalledPodman;
+      let installed = false;
       try {
         await podmanInstall.doInstallPodman(provider);
-        const installation = await getPodmanInstallation();
-        const installed = installation ? true : false;
+        installation = await getPodmanInstallation();
+        installed = installation ? true : false;
         extensionApi.context.setValue('podmanIsNotInstalled', !installed, 'onboarding');
+        telemetryLogger.logUsage('podman.onboarding.installPodman', {
+          version: installation.version || '',
+          installed: installed,
+        });
       } catch (e) {
         console.error(e);
         extensionApi.context.setValue('podmanIsNotInstalled', true, 'onboarding');
+        telemetryLogger.logError('podman.onboarding.installPodman', {
+          version: installation.version || '',
+          installed: installed,
+          error: e,
+        });
```

Comment:
we should use logUsage in both case (see some previous comments/issue)
https://github.com/containers/podman-desktop/pull/4407
https://github.com/containers/podman-desktop/issues/4214

Reply:
also probably need to use the pattern try / catch / finally (and do telemetry in finally block)